yarn-project/pxe/src/database/kv_pxe_database.ts

```diff
@@ -282,10 +308,62 @@ export class KVPxeDatabase implements PxeDatabase {
     return result;
   }
 
-  getNotes(filter: NoteFilter): Promise<IncomingNoteDao[]> {
+  getIncomingNotes(filter: IncomingNotesFilter): Promise<IncomingNoteDao[]> {
     return Promise.resolve(this.#getNotes(filter));
   }
 
+  getOutgoingNotes(filter: OutgoingNotesFilter): Promise<OutgoingNoteDao[]> {
```

Comment:
This was written by Claude AI. We could probably generalize the functionality to have 1 filter function which handles both incoming and outgoing but that would need a larger refactor of how we store notes which seemed not worth it to me.